apps/teams-test-app/src/components/PagesConfigAPIs.tsx

```diff
@@ -86,27 +86,44 @@ const SetValidityState = (): React.ReactElement =>
   });
 
 const RegisterOnRemoveHandler = (): React.ReactElement =>
-  ApiWithoutInput({
+  ApiWithTextInput({
     name: 'config_registerOnRemoveHandler',
     title: 'Register On Remove Handler',
     onClick: {
-      withPromise: async setResult => {
-        pages.config.registerOnRemoveHandler((removeEvent: pages.config.RemoveEvent): void => {
-          setResult('Remove event received.');
-          removeEvent.notifySuccess();
-        });
-        return 'config.registerOnRemoveHandler()' + noHostSdkMsg;
-      },
-      withCallback: setResult => {
-        settings.registerOnRemoveHandler((removeEvent: settings.RemoveEvent): void => {
-          setResult('Remove event received.');
-          removeEvent.notifySuccess();
-        });
+      validateInput: () => undefined,
+      submit: {
+        withPromise: async (input, setResult) => {
+          if (input) {
```

Comment:
How does this impact the existing e2e tests? Without changes to the tests that call this method, I would expect them not to work the same (and possibly fail?).

We have a test that uses this `RegisterOnRemoveHandler` element currently that is testing the `notifySuccess` path. But it doesn't expect to provide input, so we wouldn't end up calling `removeEvent.notifySuccess` like we currently do and would instead call `removeEvent.notifyFailure`. Unless I'm missing something (which is totally possible).

Reply:
This would definitely depend on the cypress test, which I believe Jess has mentioned she is making changes to currently.

Reply:
you know what, you're right, this will break all of them because they aren't looking for an input, originally I had these split into two different tests but jay suggested combining them.  I can see either being the way forward -

Option 1: I'll put the old test back so that the hub tests won't break, then make an update to the hub and finish this by making a PR to the teams sdk again. 

Option 2: split this test into a success case and a failure case then  open my PR for the hub sdk. 

How would you like to proceed?

Reply:
When I made my comment I originally thought the flow of execution would be as such:
TeamsJS SDK communicates desire to registerOnRemoveHandler. Upon removal the codepath would either pass or fail depending on if the remove is properly executed to where this SDK communicates to. From there based on the execution we would either `removeEvent.notifySuccess` or `removeEvent.notifyFailure` (i.e Remove Event has either passed or failed). 

After a discussion with Jess, I realized that this isn't what was needed and instead we would want to either force a failure or success scenario, as such I think it would be best to proceed with option 2 since they are two separate things we want to achieve rather than one.